rest-server/src/main/java/org/opennms/horizon/server/service/GrpcNodeService.java

```diff
@@ -120,7 +120,9 @@ public Mono<Boolean> discoveryByNodeIds(List<Long> ids, @GraphQLEnvironment Reso
 
     @GraphQLQuery
     public Flux<TopNNode> getTopNNode(@GraphQLEnvironment ResolutionEnvironment env, Integer timeRange, TimeRangeUnit timeRangeUnit) {
-        // see LOK-2206
-        return Flux.fromIterable(new ArrayList<>());
+        var nodes = client.listNodes(headerUtil.getAuthHeader(env));
+        var topNNodes = nodes.stream().map(nodeDTO -> nodeStatusService.getTopNNode(nodeDTO, timeRange, timeRangeUnit, env))
+            .collect(Collectors.toList());
+        return Flux.fromIterable(topNNodes).flatMap(mono -> mono);
```

Comment:
This can be simplified to skip using a stream entirely:

```suggestion
        return Flux.fromIterable(client.listNodes(headerUtil.getAuthHeader(env)))
            .flatMap(nodeDTO -> nodeStatusService.getTopNNode(nodeDTO, timeRange, timeRangeUnit, env));
```